src/reducers/movie.js

```diff
@@ -0,0 +1,68 @@
+import {
+  SET_SEARCH_FILTER,
+  SET_SORT_FILTER,
+  GET_MOVIE_DETAILS_BY_ID,
+  GET_MOVIE_DETAILS_BY_ID_SUCCESS,
+  GET_MOVIE_DETAILS_BY_ID_FAILURE,
+  GET_MOVIES_BY_SEARCH_QUERY,
+  GET_MOVIES_BY_SEARCH_QUERY_SUCCESS,
+  GET_MOVIES_BY_SEARCH_QUERY_FAILURE
+} from '../constants';
+
+const initialState = {
+  movies: [],
+  id: 0,
+  movie: [],
+  searchQuery: '',
+  searchBy: 'title',
+  sortBy: 'release_date',
+  isFetching: false
+};
+
+export const movie = (state = initialState, action) => {
+  const { type, payload } = action;
+  switch (type) {
+    case GET_MOVIES_BY_SEARCH_QUERY:
+      return {
+        ...state,
+        searchQuery: payload.search,
+        sortBy: payload.sortBy,
+        isFetching: true
+      };
+    case GET_MOVIES_BY_SEARCH_QUERY_SUCCESS:
+      return {
+        ...state,
+        movies: payload.movies,
+        isFetching: false
+      };
+    case GET_MOVIES_BY_SEARCH_QUERY_FAILURE:
+      return state;
```

Comment:
the strange case actually, if success u set isFetching: false, if failure u leave isFetching as true